libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -322,6 +347,29 @@ private async Task SendOAuthCardAsync(ITurnContext turnContext, IMessageActivity
                 {
                     signInLink = await adapter.GetOauthSignInLinkAsync(turnContext, _settings.ConnectionName, cancellationToken).ConfigureAwait(false);
                 }
+                else if (turnContext.TurnState.Get<ClaimsIdentity>("BotIdentity") is ClaimsIdentity botIdentity && SkillValidation.IsSkillClaim(botIdentity.Claims))
+                {
+                    // Generate sign in link for Skills (Emulator and WebChat don't' have the skill appId so we need to create the link for it
+                    signInLink = await adapter.GetOauthSignInLinkAsync(turnContext, _settings.ConnectionName, cancellationToken).ConfigureAwait(false);
+
+                    // In the skills preview bits, we require magic code (we'll need to update WebChat and emulator to avoid this in the next release.
+                    if (turnContext.Activity.ChannelId == Channels.Emulator || turnContext.Activity.ChannelId == Channels.Webchat)
+                    {
+                        // TODO: use the skill helper functions to handle unpacking conversation ID once we go GA with skills.
+                        // Skills need to unpack the conversation ID and create their own link.
+                        var parts = JsonConvert.DeserializeObject<string[]>(Encoding.UTF8.GetString(Convert.FromBase64String(turnContext.Activity.Conversation.Id)));

```

Comment:
This is what @Jeffders  recommended not doing IIRC?

Reply:
Emulator seems to need the conversation ID (the link doesn't open if I don't send it). I'll double check with jeff to be sure

Reply:
Updated code to only concatenate converastionId for emulator